ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -105,9 +110,34 @@ private void setupListeners() {
         .subscribe(EthPV62.NEW_BLOCK_HASHES, this::handleNewBlockHashesFromNetwork);
   }
 
+  @VisibleForTesting
+  void broadcastBlock(final Block block) {
+    blockBroadcaster = new BlockBroadcaster(ethContext);
+    final ProtocolSpec<C> protocolSpec =
+        protocolSchedule.getByBlockNumber(block.getHeader().getNumber());
+    final BlockHeaderValidator<C> blockHeaderValidator = protocolSpec.getBlockHeaderValidator();
+    final BlockHeader parent =
+        protocolContext
+            .getBlockchain()
+            .getBlockHeader(block.getHeader().getParentHash())
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        "Incapable of retrieving header from non-existent parent of "
+                            + block.getHeader().getNumber()
+                            + "."));
```

Comment:
This is ok for this PR, but could you follow up with a PR to change `importOrSavePendingBlock` so that instead of just using `protocolContext.getBlockchain().contains(block.getHeader().getParentHash())` it actually gets the parent header and checks if it's present.  Then you can pass that header into `broadcastBlock` and not need this exception.  Just be careful that the lookup of the parent header stays inside the `synchronized` block.